New Panoramax StreetView Layer (GSOC)

mattiapezzotti opened the pull request:
Brings a brand new StreetView Layer to Street Images as requested in #9941

This is still a wip, expect some bugs :P

//edit @tyrasd: For some screenshots and additional details, see https://www.openstreetmap.org/user/Juicio/diary/404174 and https://github.com/openstreetmap/iD/wiki/GSoC-2024-%E2%80%93-Panoramax-integration

mattiapezzotti committed:
added panoramax css

mattiapezzotti committed:
added mock Panoramax (mapilio clone) to photo layer list

mattiapezzotti committed:
playing with the panormax API

mattiapezzotti committed:
trying stuff with API (again) (some issues uh)

mattiapezzotti committed:
added all images (and sequences) to map as flat pictures

mattiapezzotti committed:
added modular image definition to backend

mattiapezzotti committed:
wip, nothing to see here

mattiapezzotti committed:
changed API images endpoint to be able to move in sequence (wip)

mattiapezzotti committed:
move beetween pictures in sequence

mattiapezzotti committed:
added 360 photos

mattiapezzotti committed:
added report button

mattiapezzotti committed:
hd work wip

mattiapezzotti committed:
360 photos now keep heading between sequences, added hd checkbox

mattiapezzotti committed:
added viewfield heading

mattiapezzotti committed:
reverted streetside changes (oops!)

mattiapezzotti committed:
added date and photo type filter

mattiapezzotti committed:
added draw sequences lines on lower zoom levels

mattiapezzotti committed:
refactored how panoramax shows frames

mattiapezzotti committed:
added a way to show only sequences on low level zoom

mattiapezzotti committed:
username filter wip

mattiapezzotti committed:
added username filter

mattiapezzotti committed:
added photo age filter

tyrasd committed:
fix zoom/pan glitch on plane photos

When changing to a different photo, it can be the case that the new photo has a different aspect ratio: the zoom behaviour needs to be updated.

Also, keeping the "orientation" does not make sense for plane photos, and even keeping the previously shown photo's zoom is also not typically desired for these.

mattiapezzotti committed:
Merge pull request #1 from tyrasd/panoramax

fix zoom/pan glitch on plane photos

mattiapezzotti committed:
date slider filter (kinda finished just buggy label)

mattiapezzotti committed:
Merge branch 'panoramax' of github.com:mattiapezzotti/iD into panoramax

mattiapezzotti committed:
fixed age slider label, added current selected year label and maximum oldest year

mattiapezzotti committed:
fixed typos and syntax errors

mattiapezzotti committed:
fixed npm packages

mattiapezzotti committed:
trying to fix the date slider save thing arggg

mattiapezzotti committed:
still working on the slider

mattiapezzotti committed:
the slider works now.

mattiapezzotti committed:
Merge branch 'openstreetmap:develop' into panoramax

mattiapezzotti committed:
new endpoint!

mattiapezzotti committed:
prepared for pr

mattiapezzotti committed:
added to changelog

tyrasd committed:
Merge branch 'develop' into panoramax

JackNUMBER commented:
You can set this PR as draft instead it is ready to be reviewed. 

Thanks for this contribution, can't wait to use it!

mattiapezzotti committed:
Update panoramax_images.js

Fixing a bug with heading

mattiapezzotti committed:
Update panoramax.js

Fixing a bug with heading

mattiapezzotti committed:
fixing the mess

tyrasd committed:
Merge branch 'develop' into panoramax

tyrasd committed:
switch link to panoramax.xyz viewer

tyrasd committed:
make "captured by …" translatable; single line if possible

tyrasd committed:
update photos immediately when filters change

tyrasd committed:
update username filter on input change; fix multiple names/ids

in panoramax a username can resolve to multiple ids (when the same name is used on multiple servers).
further, the username filter input field can contain more than one username

tyrasd committed:
minor tweak of changelog entry

tyrasd committed:
drop unused strings

tyrasd committed:
drop (for now) unused ui code

tyrasd committed:
lint: remove unused code, code style and early update cached username

tyrasd reviewed (approved):
@mattiapezzotti I've tested the implementation and found a few minor things that I fixed in my commits in this branch today:

* the date filters did not update immediately (this was also buggy in the mapilio layer, and likely was a simple copy-paste mistake)
* username filters: sometimes did not update when the input field was changed and did not take into account that a single user name can have more than one user id in panoramax (one per server, e.g. one for openstreetmap.fr and one for the IGN server)
* the username input field can supply more than one username
* this branch contained a bit of code (and translatable strings) which is only needed for the upcoming enhancement (date slider). I've removed it for now. You might have to pay attention to that when rebasing your other branch on the development branch after this PR is merged.

Apart from that, this seems to work quite well. Thank you!

Pujakumari1202 commented:
Hi there,

I am Puja Kumari currently planning to contribute to open source through GSOC and I found the OpenStreetMap community particularly inspiring because of its mission to create and provide free, accessible geographic data for the world.

It would be really great if the community members would help me to know regarding how should I get started.

Thanks and Regards
